main/ui/src/main/java/org/cryptomator/ui/controllers/MainController.java

```diff
@@ -216,7 +216,27 @@ public void initStage(Stage stage) {
 
 	private void gracefulShutdown() {
 		vaults.filtered(Vault.NOT_LOCKED).forEach(Vault::prepareForShutdown);
-		Platform.runLater(Platform::exit);
+		if (!vaults.filtered(Vault.NOT_LOCKED).isEmpty()) {
+			ButtonType tryAgainButtonType = new ButtonType(
+					localization.getString("main.gracefulShutdown.button.tryAgain"));
+			ButtonType forceShutdownButtonType = new ButtonType(
+					localization.getString("main.gracefulShutdown.button.forceShutdown"));
+			Alert gracefulShutdownDialog = DialogBuilderUtil.buildGracefulShutdownDialog(
+					localization.getString("main.gracefulShutdown.dialog.title"),
+					localization.getString("main.gracefulShutdown.dialog.header"),
+					localization.getString("main.gracefulShutdown.dialog.content"),
+					tryAgainButtonType,
+					forceShutdownButtonType);
+
+			Optional<ButtonType> choice = gracefulShutdownDialog.showAndWait();
+			if (choice.isPresent()) {
```

Comment:
`else` what? Will the window just close and shutdown is canceled? if not else-branch is needed, it _might_ be easier to read using 
```Java
choice.ifPresent(btnType -> {
  if (tryAgainButtonType.equals(btnType) {
    // ...
  } else if ( ... ) {
    // ...
  }
});
```

Reply:
Thank you, I've refactored it.